aten/src/ATen/core/ivalue.cpp

```diff
@@ -516,18 +517,28 @@ IValue IValue::deepcopy(
     }
       break;
     case IValue::Tag::Object: {
-      copy = IValue(toObject()->deepcopy(memo));
-      break;
+      auto class_type = type()->expect<ClassType>();
+      if (class_type->hasMethod("__getstate__") && class_type->hasMethod("__setstate__")) {
```

Comment:
Ideally all this logic would live in ivalue::Object::deepcopy, but it's difficult to recover an `IValue` instance of the object once `this` has already been unwrapped to `ivalue::Object` (intrusive_ptr ownership and constness issues). So for now this just lives here